images/metering-ansible-operator/roles/meteringconfig/tasks/configure_metastore.yml

```diff
@@ -0,0 +1,43 @@
+- name: Get metastore secret
+  block:
+    - name: Get Metastore storage credentials secret
+      k8s_info:
+        api_version: v1
+        kind: Secret
+        name: "{{ meteringconfig_spec_overrides.hive.spec.config.db.secretName }}"
+        namespace: "{{ meta.namespace }}"
+      register: metastore_secret
+  when: meteringconfig_spec_overrides.hive.spec.config.db.secretName is defined and meteringconfig_spec_overrides.hive.spec.config.db.secretName | length > 0
+
+- name: Create variables for the decoded username and password data
+  set_fact:
+    _username: "{{ metastore_secret.resources[0].data.username | b64decode }}"
```

Comment:
Since we're indexing the `data` map for both of the username and password keys, we should make a variable for the `{{ <register variable name>.resources[0].data` and reference that variable when initializing the username and password facts.